src/code.c

```diff
@@ -516,10 +516,10 @@ void PushBinaryOp (
 */
 static Int PushValue(Obj val)
 {
-    BodyHeader * header = (BodyHeader *)STATE(PtrBody);
-    if (!header->values)
-        header->values = NEW_PLIST(T_PLIST, 4);
-    return PushPlist(header->values, val);
+    Obj values = ((BodyHeader *)STATE(PtrBody))->values;
+    if (!values)
+        values = ((BodyHeader *)STATE(PtrBody))->values = NEW_PLIST(T_PLIST, 4);
```

Comment:
This line looks too clever to me (doing two assignments), also isn't PtrBody a raw pointer, so it's important if GAP calls STATE(PtrBody) before or after the RHS?

Maybe assign to `values` in this line, then assign to PtrBody in a new second line?

Reply:
You're absolutely right. Patch revised.